ocaml/xapi-idl/storage/storage_interface.ml

```diff
@@ -1285,6 +1342,16 @@ module type MIRROR = sig
     -> url:string
     -> verify_dest:bool
     -> unit
+
+  val pre_deactivate_hook :
+    context -> dbg:debug_info -> dp:dp -> sr:sr -> vdi:vdi -> unit
+
+  val is_mirror_failed :
```

Comment:
The grammar of "is...failed" is a bit odd; should this be "has..failed`?

Reply:
Sounds good, although this should be fixed in #6439